File: ambeth/Ambeth.Test/ambeth/cache/valueholdercontainer/PropertyChangeExtensionMock.cs
using De.Osthus.Ambeth.Collections;
using De.Osthus.Ambeth.Propertychange;
using System;

namespace De.Osthus.Ambeth.Cache.Valueholdercontainer
{
    public class PropertyChangeExtensionMock : IPropertyChangeExtension
    {
        protected readonly IMap<String, int> propertyNameToHitCountMap;

        public PropertyChangeExtensionMock(IMap<String, int> propertyNameToHitCountMap)
        {
            this.propertyNameToHitCountMap = propertyNameToHitCountMap;
        }

        public void PropertyChanged(Object obj, String propertyName, Object oldValue, Object currentValue)
        {
            int hitCount = propertyNameToHitCountMap.Get(propertyName);
            hitCount++;
            propertyNameToHitCountMap.Put(propertyName, hitCount);
        }
    }
}

File: jambeth/jambeth-test/src/test/java/de/osthus/ambeth/cache/valueholdercontainer/PropertyChangeExtensionMock.java
package de.osthus.ambeth.cache.valueholdercontainer;

import de.osthus.ambeth.collections.IMap;
import de.osthus.ambeth.databinding.IPropertyChangeExtension;

public class PropertyChangeExtensionMock implements IPropertyChangeExtension
{
	protected final IMap<String, Integer> propertyNameToHitCountMap;

	public PropertyChangeExtensionMock(IMap<String, Integer> propertyNameToHitCountMap)
	{
		this.propertyNameToHitCountMap = propertyNameToHitCountMap;
	}

	@Override
	public void propertyChanged(Object obj, String propertyName, Object oldValue, Object currentValue)
	{
		Integer hitCount = propertyNameToHitCountMap.get(propertyName);
		if (hitCount == null)
		{
			hitCount = Integer.valueOf(0);
		}
		hitCount++;
		propertyNameToHitCountMap.put(propertyName, hitCount);
	}
}